src/GraphQL.Tests/Instrumentation/OpenTelemetryTests.cs

```diff
@@ -0,0 +1,384 @@
+#if NET5_0_OR_GREATER
+
+using System.Diagnostics;
+using GraphQL.Instrumentation;
+using GraphQL.Types;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Trace;
+
+namespace GraphQL.Tests.Instrumentation;
+
+public class OpenTelemetryTests : IDisposable
+{
+    private readonly List<Activity> _exportedActivities = new();
+    private readonly IHost _host;
+    private readonly IDocumentExecuter<ISchema> _executer;
+    private readonly IGraphQLTextSerializer _serializer;
+    private readonly GraphQLTelemetryOptions _options;
+
+    public OpenTelemetryTests()
+    {
+        // configure services
+        _host = new HostBuilder()
+            .ConfigureServices(services =>
+            {
+                services.AddOpenTelemetry()
+                    .WithTracing(b => b
+                        .AddSource(GraphQLTelemetryProvider.SourceName) // need to specify the source name to be traced
+                        .AddInMemoryExporter(_exportedActivities));
+
+                services.AddGraphQL(b => b
+                    .AddSystemTextJson()
+                    .AddAutoSchema<Query>()
+                    .UseTelemetry());
+            })
+            .Build();
+
+        // starts telemetry services
+        _host.Start();
+
+        _executer = _host.Services.GetRequiredService<IDocumentExecuter<ISchema>>();
+        _serializer = _host.Services.GetRequiredService<IGraphQLTextSerializer>();
+        _options = _host.Services.GetRequiredService<GraphQLTelemetryOptions>();
+    }
+
+    public void Dispose() => _host.Dispose();
+
+    [Fact]
+    public async Task BasicTest()
+    {
+        // execute GraphQL document
+        var result = await _executer.ExecuteAsync(new ExecutionOptions
+        {
+            Query = "{ hello }",
+            RequestServices = _host.Services,
+        }).ConfigureAwait(false);
+
+        // verify GraphQL response
+        _serializer.Serialize(result).ShouldBe("""{"data":{"hello":"World"}}""");
+
+        // verify activity telemetry
+        var activity = _exportedActivities.ShouldHaveSingleItem();
+        activity.Tags.ShouldBe(new KeyValuePair<string, string>[]
+        {
+            new("graphql.document", "{ hello }"),
+            new("graphql.operation.type", "query"),
+            // no operation name
+        });
+        activity.DisplayName.ShouldBe("query");
+#if NET6_0_OR_GREATER
```

Comment:
What value does `activity.Status` have on NET5 or below?

Reply:
The property does not exist on NET5 and below.

Reply:
Ah.